### PR TITLE
WhitespacyLinesFixer - fix bug cases

### LINKED ISSUE #1506: whitespacy_lines removes space also from non-blank lines
It seems to me that "whitespacy_lines" removes trailing whitespace from both blank lines (containing only whitespace) and non-blank lines (what "trailing_spaces" is doing).

E.g. 
```php
<?php

$r = 5 +6;	
	
$t = true> 9;	

```

gets formatted to 

```php
<?php

$r = 5 +6;

$t = true> 9;

```

All tabs removed. Is that by design or a bug? From the description it seemed that these to formatters should not overlap.

### PR DESCRIPTION
**Pull request by keradus**
fixes #1506
replaces #1551

### CODE REVIEW

**Comment by keradus** [comment]
ping @JanisE  @SpacePossum

**Comment by keradus** [comment]
This will be the last one to v1.10.x

**Comment by SpacePossum** [comment]
lgtm :+1:  thanks

**Comment by Soullivaneuh** [comment]
:+1:

**Comment by GrahamCampbell** [comment]
:+1:

File: Symfony/CS/Fixer/Symfony/WhitespacyLinesFixer.php [hunk 1]
```diff
@@ -40,13 +40,21 @@ public function fix(\SplFileInfo $file, $content)
...
+
+                $lStart = 1;
+                if (isset($tokens[$index - 1]) && $tokens[$index - 1]->isGivenKind(T_OPEN_TAG) && "\n" === substr($tokens[$index - 1]->getContent(), -1)) {
+                    $lStart = 0;
+                }
 
-                if (isset($tokens[$index + 1])) {
-                    $newContent .= end($lines);
+                for ($l = $lStart; $l < $lMax; ++$l) {
+                    $lines[$l] = preg_replace('/^\h+$/m', '', $lines[$l]);
```
Comment (keradus) [inline Symfony/CS/Fixer/Symfony/WhitespacyLinesFixer.php]: todo: the `m` flag may not be needed anymore

Comment (keradus) [inline Symfony/CS/Fixer/Symfony/WhitespacyLinesFixer.php]: yep, removed

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**